GameOfRunes/GameOfRunes/GameEngine/Systems/MoveSystem.swift

```diff
@@ -42,16 +42,16 @@ class MoveSystem: GKComponentSystem<MoveComponent>, System {
     private func closestMoveComponent(from component: MoveComponent, for team: Team) -> GKAgent2D? {
         var closestMoveComponent: MoveComponent?
         var closestDistance: CGFloat = 0.0
-        var avoidPowerUps = false
-        
-        if let enemyType = component.entity?.component(ofType: EnemyTypeComponent.self)?.enemyType {
-            avoidPowerUps = enemyType.isPowerUpImmune
-        }
+        let isPowerUpImmune = component
+            .entity?
+            .component(ofType: EnemyTypeComponent.self)?
+            .enemyType
+            .isPowerUpImmune
```

Comment:
we should do the unpacking here so `?? false`

Reply:
Agree